Update docs related to React 18 (#35952)
vercel/next.js: docs/advanced-features/react-18/overview.md

You can now start using React 18's new APIs like `startTransition` and `Suspense` in Next.js.

## Streaming SSR

> Next.js supports React 18's Streaming server-rendering (SSR) out of the box for SSR pages.

Reviewer: ```suggestion
Next.js supports React 18 streaming server-rendering (SSR) out of the box.
```

[Learn more about streaming in Next.js](/docs/advanced-features/react-18/streaming.md).

## React Server Components (Alpha)

Server Components are a new feature in React that let you reduce your JavaScript bundle size by separating server and client-side code. Server Components allow developers to build apps that span the server and client, combining the rich interactivity of client-side apps with the improved performance of traditional server rendering.

Server Components are still in research and development. [Learn how to try Server Components](/docs/advanced-features/react-18/server-components.md) as an experimental feature in Next.js.

## Switchable Runtime (Alpha)

> Next.js supports changing the runtime of your application between Node.js and [Edge Runtime](/docs/api-reference/edge-runtime), in a per-page level. For example, you can selectively configure some specific pages to be server-side rendered in the Edge Runtime.

Reviewer 2: ```suggestion
Next.js supports changing the runtime of your application between Node.js and [Edge Runtime](/docs/api-reference/edge-runtime.md), at the page level. For example, you can selectively configure specific pages to be server-side rendered in the Edge Runtime.
```

> This feature is still experimental. [Learn more about switchable runtime](/docs/advanced-features/react-18/switchable-runtime.md).

Reviewer: ```suggestion
This feature is still experimental. [Learn more about the switchable runtime](/docs/advanced-features/react-18/switchable-runtime.md).
```
